pyvrp/read.py

```diff
@@ -552,3 +552,40 @@ def _allowed2profile(self) -> dict[tuple[int, ...], int]:
                 allowed_clients2profile_idx[clients] = next(profile_idx)
 
         return allowed_clients2profile_idx
+
+
+def _problem_data_from_json(cls, data: dict) -> ProblemData:
```

Comment:
This doesn't do what it says: it parses a dict, not a JSON string. Presumably the dict has previously been parsed from a JSON-like string?

A `from_json`-like interface should probably have the following signature:
```python
def from_json(cls, json_str: str) -> ProblemData
```
with intended usage
```python
with open("data.json") as fh:
    data = ProblemData.from_json(fh.read())
```